index.bs

```diff
@@ -87,8 +87,40 @@ never be allowed using this API, rather than leaving it entirely up to underlyin
 systems.
 
 A <dfn lt="file|file entry">file entry</dfn> additionally consists of
-<dfn for="file entry">binary data</dfn> (a [=byte sequence=]) and a
-<dfn for="file entry">modification timestamp</dfn> (a number representing the number of milliseconds since the <a spec=FileAPI>Unix Epoch</a>).
+<dfn for="file entry">binary data</dfn> (a [=byte sequence=]), a
+<dfn for="file entry">modification timestamp</dfn> (a number representing the number of milliseconds since the <a spec=FileAPI>Unix Epoch</a>)
+and a <dfn for="file entry">lock</dfn>. A lock is a string that may exclusively be "`open`", "`taken-exclusive`", and "`taken-shared`".
+
+<div algorithm>
+To <dfn for="file entry/lock">take</dfn> a [=file entry/lock=] with a |value| of "`exclusive`" or "`shared`" on a given [=file entry=] |file|,
+run the following steps:
+
+1. Let |lock| be the |file|'s [=file entry/lock=].
+1. If |value| is "`exclusive`":
+   1. If |lock| is "`open`":
+     1. Set lock to "`taken-exclusive`".
+     1. Return `true`.
```

Comment:
Style nit: we don't use markup for boolean values, see https://infra.spec.whatwg.org/#booleans.